dds/idl/field_info.h

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef anonymous_H
+#define anonymous_H
+
+#include "dds_generator.h"
+#include <utl_scoped_name.h>
+#include <ast.h>
+
+#include <string>
+
+struct FieldInfo {
+  struct EleLen {
+    AST_Type* ele_;
+    std::size_t len_;
```

Comment:
I'm not sure this is correct for multidimensional arrays.  Is there a need to distinguish among `octet[2][3]`, `octet[3][2]`, `octet[6]`, etc.?  Please add some of these kinds of cases to the test.

Reply:
The screening set EleLenSet is not used for anonymous arrays. It's true that EleLenSet would be incorrect to screen multidimensional anonymous arrays. We would need something like a string to store 2x3 or 3x2, etc. Right now EleLenSet is only used to screen anonymous sequences. I will change the variable name accordingly as you pointed out in another comment.

Reply:
Got it, I must have misread that part of how it's used.